Sources/ArcGISToolkit/Components/Offline/OfflineMapAreasView.MapViewModel.swift

```diff
@@ -55,10 +67,93 @@ extension OfflineMapAreasView {
             }
         }
         
-        /// Requests authorization to show notifications.
-        nonisolated func requestUserNotificationAuthorization() async {
-            _ = try? await UNUserNotificationCenter.current()
-                .requestAuthorization(options: [.alert, .sound])
+        /// Makes offline preplanned map models with information from the downloaded mobile map
+        /// packages for the online map.
+        func makeOfflinePreplannedMapModels() async {
+            guard let portalItemID else { return }
+            
+            let preplannedDirectory = FileManager.default.preplannedDirectory(forPortalItemID: portalItemID)
+            
+            guard let mapAreaIDs = try? FileManager.default.contentsOfDirectory(atPath: preplannedDirectory.path()) else { return }
+            
+            var mapAreas: [OfflinePreplannedMapArea] = []
+            
+            for mapAreaID in mapAreaIDs {
+                if let preplannedMapAreaID = PortalItem.ID(mapAreaID),
+                   let mapArea = await createMapArea(
+                    for: portalItemID,
+                    preplannedMapAreaID: preplannedMapAreaID
+                   ) {
+                    mapAreas.append(mapArea)
+                }
+            }
+            
+            offlinePreplannedMapModels = mapAreas.map { mapArea in
+                PreplannedMapModel(
+                    offlineMapTask: offlineMapTask,
+                    mapArea: mapArea,
+                    portalItemID: portalItemID,
+                    preplannedMapAreaID: mapArea.id!
+                )
+            }
+            .sorted(using: KeyPathComparator(\.preplannedMapArea.title))
```

Comment:
I see a sendability warning on this line in Xcode 16.0, it seems that it will be gone when the upcoming feature becomes formal though, so no need to fix: https://github.com/swiftlang/swift/issues/57560